### PR TITLE
add APIServer config type

### PR DESCRIPTION
**Pull request by sanchezl**
Add APIServer config type. 
Add the ability to specify client CA.
Add the ability to specify default and named certs.
```
---
kind: APIServer
spec:
  clientCA:
    name: baz
  servingCerts:
    defaultServingCertificate:
      name: name_default
    namedCertificates:
    - servingCertificate:
        name: name_foo
    - servingCertificate:
        name: name_bar
      names:
      - bar.com
      - *.bar.com
```

### CODE REVIEW

File: config/v1/types_apiserver.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+// APIServer holds cluster-wide information about api-servers.  The canonical name is `cluster`
+type APIServer struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec APIServerSpec `json:"spec"`
+	Status APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: @sanchezl do you intend for this to include the cluster-admin provided client-ca for the kube-apiserver?

@sttts @mfojtik

File: config/v1/types_apiserver.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIServer holds cluster-wide information about api-servers.  The canonical name is `cluster`
+type APIServer struct {
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: @openshift/api-reviewers  let's argue over names and what is included.

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (deads2k) [inline config/v1/types_apiserver.go]: @sttts would you place audit in here?  It's a possibility

File: config/v1/types_apiserver.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec APIServerSpec `json:"spec"`
```
Comment (sttts) [inline config/v1/types_apiserver.go]: @deads2k this is also the place for the audit config, right?

File: config/v1/types_apiserver.go (hunk 2 above)
Comment (adambkaplan) [reply config/v1/types_apiserver.go]: Is this for the `kube-apiserver`, `openshift-apiserver`, or both?

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (sttts) [reply config/v1/types_apiserver.go]: This is my understanding. But then I don't understand e.g. NamedCertificates or APIServerCertInfo. We have to carefully split the config we have today.

File: config/v1/types_apiserver.go (hunk 2 above)
Comment (sttts) [reply config/v1/types_apiserver.go]: GenericAPIServerConfig

Comment (sanchezl) [reply config/v1/types_apiserver.go]: * There is already an [existing `GenericAPIServerConfig` type](https://github.com/openshift/api/blob/d22df4f5cec775b9d689711da4d193766aa50dfe/config/v1/types.go#L222).
* None of the other cluster config types end in `Config`
* Some more suggestions:
  * `APIServerTLS` (if TLS certs are all that end up in here)
  * `APIServerCommon`
  * `APIServers`

Comment (soltysh) [reply config/v1/types_apiserver.go]: :+1: for the current name, it's short and clean.

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: * I hear audit config will be (at some point, possibly post-4.0) in its own config. (@deads2k / @smarterclayton might have more info)
* `APIServerCertInfo` & `NameCertificates` are only used by the kube-apiserver, who has exposed endpoints.

File: config/v1/types_apiserver.go (hunk 2 above)
Comment (deads2k) [reply config/v1/types_apiserver.go]: > Is this for the `kube-apiserver`, `openshift-apiserver`, or both?

KubeAPIServer, but I'm ok claiming the name `APIServer` here.



> * `APIServerTLS` (if TLS certs are all that end up in here)
> * `APIServerCommon`
> * `APIServers`

tls is too specific, common is just another way of saying stuff, plural implies a thing we won't do

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (deads2k) [reply config/v1/types_apiserver.go]: > This is my understanding. But then I don't understand e.g. NamedCertificates or APIServerCertInfo. We have to carefully split the config we have today.

Having given this more thought, I think audit and admission would be separate and cross cutting.  Audit may be dropped in the future.

File: config/v1/types_apiserver.go (hunk 3 above)
Comment (deads2k) [reply config/v1/types_apiserver.go]: > @deads2k this is also the place for the audit config, right?

I think we'll have a separate top level audit.

File: config/v1/types_apiserver.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+type APIServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// APIServerCertInfo is the default TLS cert info for serving secure traffic.
+	APIServerCertInfo `json:",inline"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: This is hard to use. Just name it as the `DefaultServingCert`

File: config/v1/types_apiserver.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	// NamedCertificates is a list of certificates to use to secure requests to specific hostnames
+	NamedCertificates []APIServerNamedCertificate `json:"namedCertificates"`
+	// ClientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
+	ClientCA ConfigMapNameReference `json:"clientCA"`
+}
+
+type APIServerCertInfo struct {
+	CertFile ConfigMapNameReference `json:"certFile"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: no.  We want to reference  tls type secret that has tls.key and tls.cert

Comment (deads2k) [reply config/v1/types_apiserver.go]: > no. We want to reference tls type secret that has tls.key and tls.cert

it's a core kube type of secret

File: config/v1/types_apiserver.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+type APIServerSpec struct {
+	// APIServerCertInfo is the default TLS cert info for serving secure traffic.
+	APIServerCertInfo `json:",inline"`
+	// NamedCertificates is a list of certificates to use to secure requests to specific hostnames
+	NamedCertificates []APIServerNamedCertificate `json:"namedCertificates"`
+	// ClientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
+	ClientCA ConfigMapNameReference `json:"clientCA"`
+}
+
+type APIServerCertInfo struct {
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: which makes this unnecessary

File: config/v1/types_apiserver.go (hunk 4 above)
Comment (deads2k) [inline config/v1/types_apiserver.go]: document what happens when empty.  mark as optional, use the secretref type

File: config/v1/types_apiserver.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+type APIServerSpec struct {
+	// ServingCert is the TLS cert info for serving secure traffic.
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitEmpty"`
+	// ClientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
+	ClientCA ConfigMapNameReference `json:"clientCA,omitempty"`
+}
+
+type APIServerServingCerts struct {
+	// Default references a secret containing the default TLS cert info for serving secure traffic.
+	Default corev1.SecretReference `json:"default,omitempty"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: add the funny +optional thing

File: config/v1/types_apiserver.go [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+	// Named references secrets containing the TLS cert info for serving secure traffic to specific hostnames.
+	Named []APIServerNamedServingCert `json:"named,omitempty"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: optional

File: config/v1/types_apiserver.go [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// ServingCert is the TLS cert info for serving secure traffic.
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitEmpty"`
+	// ClientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: lowercase to match json

File: config/v1/types_apiserver.go [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+type APIServerServingCerts struct {
+	// Default references a secret containing the default TLS cert info for serving secure traffic.
+	Default corev1.SecretReference `json:"default,omitempty"`
+	// Named references secrets containing the TLS cert info for serving secure traffic to specific hostnames.
+	Named []APIServerNamedServingCert `json:"named,omitempty"`
+}
+
+type APIServerNamedServingCert struct {
+	// Names is a list of DNS names this certificate should be used to secure
+	// A name can be a normal DNS name, or can contain leading wildcard segments.
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: at least one name is required rigth?

File: config/v1/types_apiserver.go [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+	Names []string `json:"names"`
+	// Secret references the secret containing the TLS cert info for serving secure traffic.
+	Secret corev1.SecretReference `json:"secret"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: indicate namespace

File: config/v1/types_apiserver.go [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+type APIServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// ServingCert is the TLS cert info for serving secure traffic.
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitEmpty"`
```
Comment (mfojtik) [inline config/v1/types_apiserver.go]: @deads2k do we want omitempty on config types? (also it is `omitempty`)

File: config/v1/types_apiserver.go [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+type APIServerServingCerts struct {
+	// Default references a secret containing the default TLS cert info for serving secure traffic.
+	Default corev1.SecretReference `json:"default,omitempty"`
+	// Named references secrets containing the TLS cert info for serving secure traffic to specific hostnames.
+	Named []APIServerNamedServingCert `json:"named,omitempty"`
+}
+
+type APIServerNamedServingCert struct {
```
Comment (mfojtik) [inline config/v1/types_apiserver.go]: nit: probably deserves a godoc

File: config/v1/types_apiserver.go [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+	// Names is a list of DNS names this certificate should be used to secure
+	// A name can be a normal DNS name, or can contain leading wildcard segments.
+	Names []string `json:"names"`
+	// Secret references the secret containing the TLS cert info for serving secure traffic.
+	Secret corev1.SecretReference `json:"secret"`
+}
+
+type APIServerStatus struct {
```
Comment (mfojtik) [inline config/v1/types_apiserver.go]: @deads2k @sanchezl do we want to report anything here? or just have to be here for consistency?

File: config/v1/types_apiserver.go (hunk 7 above)
Comment (mfojtik) [reply config/v1/types_apiserver.go]: maybe `DefaultSecretName` ?

Comment (mfojtik) [reply config/v1/types_apiserver.go]: (i think this needs a better name)

File: config/v1/types_apiserver.go (hunk 13 above)
Comment (soltysh) [reply config/v1/types_apiserver.go]: Definitely deserves a godoc.

Comment (sanchezl) [reply config/v1/types_apiserver.go]: :+1:

File: config/v1/types_apiserver.go (hunk 11 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: Done

File: config/v1/types_apiserver.go [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// ServingCert is the TLS cert info for serving secure traffic.
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitEmpty"`
+	// ClientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
+	ClientCA ConfigMapNameReference `json:"clientCA,omitempty"`
```
[the remark this answers is not in the document]
Comment (deads2k) [reply config/v1/types_apiserver.go]: > Done

forgot to push?

File: config/v1/types_apiserver.go (hunk 7 above)
Comment (deads2k) [reply config/v1/types_apiserver.go]: > maybe `DefaultSecretName` ?

that doesn't provide more information

`DefaultServingCert` could work.

File: config/v1/types_apiserver.go (hunk 14 above)
Comment (deads2k) [reply config/v1/types_apiserver.go]: > @deads2k @sanchezl do we want to report anything here? or just have to be here for consistency?

consistency for now

File: config/v1/types_apiserver.go (hunk 15 above)
[the remark this answers is not in the document]
Comment (deads2k) [reply config/v1/types_apiserver.go]: Still missing the namespace.  `openshift-config` namespace

File: config/v1/types_apiserver.go (hunk 7 above)
Comment (deads2k) [reply config/v1/types_apiserver.go]: @sanchezl I defer to @mfojtik 's request for a more specific name.  `DefaultServingCert`

File: config/v1/types_apiserver.go [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+
+// APIServerNamedServingCert maps a server DNS name, as understood by a client, to a certificate.
+type APIServerNamedServingCert struct {
+	// names is a optional list of explicit DNS names (leading wildcards allowed) that should use this certificate to
+	// serve secure traffic. If no names are provided, the implicit names will be extracted from the certificates.
+	// Exact names trump over wildcard names. Explicit names defined here trump over extracted implicit names.
+	// +optional
+	Names []string `json:"names"`
+	// secret references a secret in a namespace containing the TLS cert info for serving secure traffic.
+	Secret corev1.SecretReference `json:"secret"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: `secret`?  I know it's a secret from type.  This is a `ServingCert` right?

File: config/v1/types_apiserver.go [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// APIServerNamedServingCert maps a server DNS name, as understood by a client, to a certificate.
+type APIServerNamedServingCert struct {
+	// names is a optional list of explicit DNS names (leading wildcards allowed) that should use this certificate to
+	// serve secure traffic. If no names are provided, the implicit names will be extracted from the certificates.
+	// Exact names trump over wildcard names. Explicit names defined here trump over extracted implicit names.
+	// +optional
+	Names []string `json:"names"`
+	// secret references a secret in a namespace containing the TLS cert info for serving secure traffic.
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: indicate namespace this secret is in.

File: config/v1/types_apiserver.go [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+	// clientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
+	// +optional
+	ClientCA ConfigMapNameReference `json:"clientCA,omitempty"`
+}
+
+type APIServerServingCerts struct {
+	// default references a secret in a namespace containing the default TLS cert info for serving secure traffic.
+	// If no named certificates match the server name as understood by a client, this default certificate will be used.
+	// +optional
+	Default corev1.SecretReference `json:"default,omitempty"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: local secret reference

File: config/v1/types_apiserver.go [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// servingCert is the TLS cert info for serving secure traffic.
+	// +optional
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitempty"`
+	// clientCA is the certificate bundle for all the signers that you'll recognize for incoming client certificates
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: that you'll recognize for incoming client certifiates in addition too cluster maintained signers

File: config/v1/types_apiserver.go (hunk 15 above)
[the remark this answers is not in the document]
Comment (sanchezl) [reply config/v1/types_apiserver.go]: Sorry, incomplete push.

File: config/v1/types_apiserver.go [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+type APIServerServingCerts struct {
+	// default references a kubernetes.io/tls type secret containing the default TLS cert info for serving secure traffic.
+	// If no named certificates match the server name as understood by a client, this default certificate will be used.
+	// The secret must exist in the openshift-config namespace and contain the following required fields:
+	// - Secret.Data["tls.key"] - TLS private key.
+	// - Secret.Data["tls.crt"] - TLS certificate.
+	// +optional
+	Default SecretNameReference `json:"default,omitempty"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: This is the name that @mfojtik asked to change.  `defaultServingCertificate` I think

File: config/v1/types_apiserver.go [hunk 21]
```diff
@@ -0,0 +1,67 @@
...
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// servingCert is the TLS cert info for serving secure traffic.
+	// +optional
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitempty"`
+	// clientCA is the certificate bundle for the signers that will be recognized for incoming client certificates.
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: for incoming client certificates in addition to the system managed client certificates.  If this is empty, then only system managed certificates are valid.  You usually only have to set this if you have your own PKI you to honor client certificates from

File: config/v1/types_apiserver.go [hunk 22]
```diff
@@ -0,0 +1,67 @@
...
+	// The namespace for this config map is openshift-config.
+	// +optional
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: @openshift/api-reviewers if you ever see this tag, demand that they explain what happens if it isn't set.

File: config/v1/types_apiserver.go [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+type APIServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// servingCert is the TLS cert info for serving secure traffic.
+	// +optional
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: What if this is empty?

File: config/v1/types_apiserver.go [hunk 24]
```diff
@@ -0,0 +1,67 @@
...
+	ClientCA ConfigMapNameReference `json:"clientCA,omitempty"`
+}
+
+type APIServerServingCerts struct {
+	// default references a kubernetes.io/tls type secret containing the default TLS cert info for serving secure traffic.
+	// If no named certificates match the server name as understood by a client, this default certificate will be used.
+	// The secret must exist in the openshift-config namespace and contain the following required fields:
+	// - Secret.Data["tls.key"] - TLS private key.
+	// - Secret.Data["tls.crt"] - TLS certificate.
+	// +optional
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: What happens if this is empty.

File: config/v1/types_apiserver.go (hunk 21 above)
Comment (deads2k) [inline config/v1/types_apiserver.go]: you're requiring ca-bundle.crt

File: config/v1/types_apiserver.go (hunk 8 above)
Comment (deads2k) [reply config/v1/types_apiserver.go]: `namedCertificates`

File: config/v1/types_apiserver.go [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+	// names is a optional list of explicit DNS names (leading wildcards allowed) that should use this certificate to
+	// serve secure traffic. If no names are provided, the implicit names will be extracted from the certificates.
+	// Exact names trump over wildcard names. Explicit names defined here trump over extracted implicit names.
+	// +optional
+	Names []string `json:"names"`
+	// secret references a kubernetes.io/tls type secret containing the TLS cert info for serving secure traffic.
+	// The secret must exist in the openshift-config namespace and contain the following required fields:
+	// - Secret.Data["tls.key"] - TLS private key.
+	// - Secret.Data["tls.crt"] - TLS certificate.
+	Secret SecretNameReference `json:"secret"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: `servingCertificate`

File: config/v1/types_apiserver.go (hunk 23 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: Update doc.

File: config/v1/types_apiserver.go (hunk 22 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: Updated doc.

File: config/v1/types_apiserver.go (hunk 24 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: Updated doc.

File: config/v1/types_apiserver.go (hunk 8 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: renamed

File: config/v1/types_apiserver.go [hunk 26]
```diff
@@ -0,0 +1,73 @@
...
+// APIServer holds cluster-wide information about api-servers.  The canonical name is `cluster`
+type APIServer struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              APIServerSpec   `json:"spec"`
+	Status            APIServerStatus `json:"status"`
+}
+
+type APIServerSpec struct {
+	// servingCert is the TLS cert info for serving secure traffic. If not specified, system managed certificates
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: I misspoke.  "operator managed certificates"  System managed certs means something different

File: config/v1/types_apiserver.go [hunk 27]
```diff
@@ -0,0 +1,73 @@
...
+	// will be used for serving secure traffic.
+	// +optional
+	ServingCerts APIServerServingCerts `json:"servingCerts,omitempty"`
+	// clientCA references a ConfigMap containing a certificate bundle for the signers that will be recognized for
+	// incoming client certificates in addition to the system managed signers. If this is empty, then only system managed signers are valid.
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: operator managed

again, my mistake.  I didn't see it until I read it.

File: config/v1/types_apiserver.go [hunk 28]
```diff
@@ -0,0 +1,73 @@
...
+	// names is a optional list of explicit DNS names (leading wildcards allowed) that should use this certificate to
+	// serve secure traffic. If no names are provided, the implicit names will be extracted from the certificates.
+	// Exact names trump over wildcard names. Explicit names defined here trump over extracted implicit names.
+	// +optional
+	Names []string `json:"names"`
+	// servingCertificate references a kubernetes.io/tls type secret containing the TLS cert info for serving secure traffic.
+	// The secret must exist in the openshift-config namespace and contain the following required fields:
+	// - Secret.Data["tls.key"] - TLS private key.
+	// - Secret.Data["tls.crt"] - TLS certificate.
+	servingCertificate SecretNameReference `json:"servingCertificate"`
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: must be public

File: config/v1/types_apiserver.go [hunk 29]
```diff
@@ -0,0 +1,73 @@
...
+	// The ConfigMap must exist in the openshift-config namespace and contain the following required fields:
+	// - ConfigMap.Data["ca-bundle.crt"] - CA bundle.
+	// +optional
+	ClientCA ConfigMapNameReference `json:"clientCA,omitempty"`
+}
+
+type APIServerServingCerts struct {
+	// defaultServingCertificate references a kubernetes.io/tls type secret containing the default TLS cert info for
+	// serving secure traffic. If no named certificates match the server name as understood by a client, this default
+	// certificate will be used. If defaultServingCertificate is not specified, then a system managed certificate will
```
Comment (deads2k) [inline config/v1/types_apiserver.go]: then an operator managed certificate

File: config/v1/types_apiserver.go (hunk 27 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: np. updated doc.

File: config/v1/types_apiserver.go (hunk 26 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: updated doc.

File: config/v1/types_apiserver.go (hunk 25 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: updated

File: config/v1/types_apiserver.go (hunk 21 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: updated doc.

File: config/v1/types_apiserver.go (hunk 29 above)
Comment (sanchezl) [reply config/v1/types_apiserver.go]: updated doc.

**Pull request merged**